### PR TITLE
Feature: integration

### PR DESCRIPTION
**Pull request by moiskillnadne**

### CODE REVIEW

File: src/widgets/ActivityGroups/model/hooks/useIntegrationsSync.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export const useIntegrationsSync = ({ appletDetails }: Props) => {
+  const dispatch = useAppDispatch()
+
+  const isLorisIntegrationAvailable = appletDetails.integrations.some(integration =>
+    AVAILABLE_INTEGRATIONS.includes(integration),
+  )
+
+  if (isLorisIntegrationAvailable) {
+    dispatch(actions.applyDataSharingSettings({ appletId: appletDetails.id }))
+  }
```
Comment (BamMironov) [inline src/widgets/ActivityGroups/model/hooks/useIntegrationsSync.ts:20]: Please use AppletIntegrationsService class. This way, you will not have to check for a specific integration.
```
useEffect(() => {
  appletIntegrationService.applyIntegrations(appletDetails);
}, [appletDetails, appletIntegrationService]);
```

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/widgets/ActivityGroups/model/hooks/useIntegrationsSync.ts (hunk 1 above)
Comment (moiskillnadne) [reply src/widgets/ActivityGroups/model/hooks/useIntegrationsSync.ts]: Done

File: src/widgets/ActivityGroups/model/hooks/useIntegrationsSync.ts [hunk 2]
```diff
@@ -12,13 +14,15 @@ export const useIntegrationsSync = ({ appletDetails }: Props) => {
...
     AVAILABLE_INTEGRATIONS.includes(integration),
   )
 
-  if (isLorisIntegrationAvailable) {
-    appletIntegrationService.applyIntegrations(appletDetails)
-  }
+  useEffect(() => {
+    const appletIntegrationService = new AppletIntegrationsService(rootState, dispatch)
+
+    if (isLorisIntegrationAvailable) {
```
Comment (BamMironov) [inline src/widgets/ActivityGroups/model/hooks/useIntegrationsSync.ts:24]: Please leave out if block and isLorisIntegrationAvailable variable

Comment (moiskillnadne) [reply src/widgets/ActivityGroups/model/hooks/useIntegrationsSync.ts]: Done

**Review by BamMironov — APPROVED** [review]

**Pull request merged**